lib/msf/base/serializer/opt_conditions.rb

```diff
@@ -0,0 +1,52 @@
+# -*- coding: binary -*-
+module Msf
+  module Serializer
+    module OptConditions
+
+      # Check a condition's result
+      def self.eval_condition(left_value, operator, right_value)
+        result = true
+        case operator
+        when "=="
+          result = left_value == right_value
+        when "!="
+          result = left_value != right_value
+        when "in"
+          result = right_value.include?(left_value)
+        when "nin"
+          result = right_value.include?(left_value)
+          result = !result
+        end
+
+        result
+      end
+
+      # Check an OPTION conditions. This function supports
+      # dump_options()
+      def self.show_option(mod, opt)
+        result = true
+        condition = opt.conditions
+
+        return true if condition.nil? || condition.empty?
+        operator = condition[1]
+        right_value = condition[2]
+
+        if condition[0][0..5] == "OPTION"
+          left_name = condition[0].split(/\(([^)]+)\)/)[1]
+          left_value = mod.datastore[left_name].nil? ? opt.default : mod.datastore[left_name]
+          result = eval_condition(left_value, operator, right_value)
+
+        elsif condition != [] and condition[0] == "ACTION"
+          left_value = mod.action.name.to_s
+          result = eval_condition(left_value, operator, right_value)
+
+        elsif condition != [] and condition[0] == "TARGET"
+          left_value = mod.target.name.to_s
+          result = eval_condition(left_value, operator, right_value)
+        end
+        result
+      end
```

Comment:
A few things about this method as well, in addition to the boolean keywords mentioned above. Like the previous method, we don't need to keep track of a result variable here, instead we could use the output of the `if`, or even better, refactor the method so that their is one `eval_condition` call and it is the last thing the method does. `condition` will not ever be the empty array in the `if` since we return early when it is `empty?`. Also, we do not need to make a special syntax for options, we can instead process options as the default case. It is also possible to make the API such that `opt.conditions` will not be `nil` and having that restriction further up the stack rather than defensive programming everywhere makes the API easier to use. Finally, Ruby's `||` operator doesn't return a boolean value all the time, it actually returns the first argument if it is not `nil` or `false` and the second argument otherwise. We can use it to make the ternary shorter and more idiomatic Ruby.

```suggestion
      def self.show_option(mod, opt)
        return true if opt.conditions.empty?

        left_source = opt.condtions[0]
        operator = opt.conditions[1]
        right_value = opt.conditions[2]

        if left_source == "ACTION"
          left_value = mod.action.name.to_s
        elsif left_source == "TARGET"
          left_value = mod.target.name.to_s
        else
          left_value = mod.datastore[left_source] || opt.default
        end
        
        eval_condition(left_value, operator, right_value)
      end
```

Reply:
Thank you very much! That was a very good idea to optimize and ease the code :)